### ISSUE TITLE
LuaJIT + sol2 linker errors with STB libraries

### ISSUE THREAD

**Issue by TasmanianNoob** [microsoft/vcpkg#35502]
**Describe the bug**
When including the STB libraries in a project that uses LuaJIT through sol2 I get linker errors. Removing the STB libraries fixes the issue. Here's my code, CMakeLists.txt and vcpkg.json files. The vcpkg file has the exact version I'm currently using for each libraries.
`main.cpp`
```
#define SOL_ALL_SAFETIES_ON 1
#include "sol/sol.hpp"

int main()
{
	sol::state lua;
	lua.open_libraries(sol::lib::base, sol::lib::string, sol::lib::table, sol::lib::math);

	lua.script("print('Hello, World!')");

	return 0;
}
```
`CMakeLists.txt`
```
cmake_minimum_required(VERSION 3.20)
project(CppTesting)

set(CMAKE_CXX_STANDARD 20)

add_executable(CppTesting main.cpp)

find_package(PkgConfig REQUIRED)
pkg_check_modules(LuaJIT REQUIRED IMPORTED_TARGET GLOBAL luajit)
target_link_libraries(${PROJECT_NAME} PRIVATE PkgConfig::LuaJIT)

find_package(sol2 CONFIG REQUIRED)
target_link_libraries(${PROJECT_NAME} PRIVATE sol2)

find_package(Stb REQUIRED)
target_include_directories(${PROJECT_NAME} PRIVATE ${Stb_INCLUDE_DIR})
```
`vcpkg.json`
```
{
  "name" : "cpptesting",
  "version-string" : "1.0.0",
  "builtin-baseline" : "13c3c0fcc203d179f4443fe48d252e3ff220cbeb",
  "dependencies" : [ {
    "name" : "vcpkg-cmake-config",
    "version>=" : "2022-02-06#1"
  }, {
    "name" : "vcpkg-cmake",
    "version>=" : "2023-05-04"
  }, {
    "name" : "pkgconf",
    "version>=" : "1.8.0#5"
  }, {
    "name" : "luajit",
    "version>=" : "2023-01-04#4"
  }, {
    "name" : "stb",
    "version>=" : "2023-04-11#1"
  }, {
    "name" : "sol2",
    "version>=" : "3.3.0"
  } ]
}
```

**Environment**
- OS: Windows 11 Home
- Compiler: MSVC 19.36.32535
- Vcpkg: vcpkg package management program version 2023-10-18-27de5b69dac4b6fe8259d283cd4011e6d20a84ce

**To Reproduce**
1. Create an empty CLion project.
2. Install sol2, pkgconf, luajit and stb with vcpkg.
3. Use vcpkg manifest.
4. Set main.cpp and CMakeLists.txt to the provided examples and compile the project.
5. The linker fails.

**Expected behavior**
I expect console to output:
`Hello, World!`

**Failure logs**
```[1/1] Linking CXX executable CppTesting.exe
FAILED: CppTesting.exe 
cmd.exe /C "cd . && C:\Users\charl\AppData\Local\JetBrains\Toolbox\apps\CLion\ch-0\232.9921.42\bin\cmake\win\x64\bin\cmake.exe -E vs_link_exe --intdir=CMakeFiles\CppTesting.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100220~1.0\x86\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100220~1.0\x86\mt.exe --manifests  -- C:\PROGRA~1\MICROS~3\2022\COMMUN~1\VC\Tools\MSVC\1436~1.325\bin\Hostx86\x64\link.exe /nologo CMakeFiles\CppTesting.dir\main.cpp.obj  /out:CppTesting.exe /implib:CppTesting.lib /pdb:CppTesting.pdb /version:0.0 /machine:x64 /debug /INCREMENTAL /subsystem:console  vcpkg_installed\x64-windows\debug\lib\lua51.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib && cmd.exe /C "cd /D C:\Users\charl\CLionProjects\CppTesting\cmake-build-debug && C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -noprofile -executionpolicy Bypass -file C:/Users/charl/.vcpkg-clion/vcpkg/scripts/buildsystems/msbuild/applocal.ps1 -targetBinary C:/Users/charl/CLionProjects/CppTesting/cmake-build-debug/CppTesting.exe -installedDir C:/Users/charl/CLionProjects/CppTesting/cmake-build-debug/vcpkg_installed/x64-windows/debug/bin -OutVariable out""
LINK Pass 1: command "C:\PROGRA~1\MICROS~3\2022\COMMUN~1\VC\Tools\MSVC\1436~1.325\bin\Hostx86\x64\link.exe /nologo CMakeFiles\CppTesting.dir\main.cpp.obj /out:CppTesting.exe /implib:CppTesting.lib /pdb:CppTesting.pdb /version:0.0 /machine:x64 /debug /INCREMENTAL /subsystem:console vcpkg_installed\x64-windows\debug\lib\lua51.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:CMakeFiles\CppTesting.dir/intermediate.manifest CMakeFiles\CppTesting.dir/manifest.res" failed (exit code 1120) with the following output:
main.cpp.obj : error LNK2019: unresolved external symbol __imp_lua_absindex referenced in function "public: decltype(auto) __cdecl sol::basic_protected_function<class sol::stack_reference,1,class sol::basic_reference<0> >::call<>(void)const " (??$call@$$V$$Z$$V@?$basic_protected_function@Vstack_reference@sol@@$00V?$basic_reference@$0A@@2@@sol@@QEBA?A_TXZ)
main.cpp.obj : error LNK2019: unresolved external symbol __imp_lua_rotate referenced in function "public: __cdecl sol::detail::protected_handler<1,class sol::basic_reference<0> >::~protected_handler<1,class sol::basic_reference<0> >(void)" (??1?$protected_handler@$00V?$basic_reference@$0A@@sol@@@detail@sol@@QEAA@XZ)
main.cpp.obj : error LNK2019: unresolved external symbol __imp_lua_getglobal referenced in function "class sol::basic_reference<0> __cdecl sol::detail::get_default_handler<class sol::basic_reference<0>,0>(struct lua_State *)" (??$get_default_handler@V?$basic_reference@$0A@@sol@@$0A@@detail@sol@@YA?AV?$basic_reference@$0A@@1@PEAUlua_State@@@Z)
main.cpp.obj : error LNK2019: unresolved external symbol __imp_lua_setglobal referenced in function "void __cdecl sol::detail::set_default_handler<class sol::basic_object<class sol::basic_reference<0> > >(struct lua_State *,class sol::basic_object<class sol::basic_reference<0> > const &)" (??$set_default_handler@V?$basic_object@V?$basic_reference@$0A@@sol@@@sol@@@detail@sol@@YAXPEAUlua_State@@AEBV?$basic_object@V?$basic_reference@$0A@@sol@@@1@@Z)
main.cpp.obj : error LNK2019: unresolved external symbol __imp_lua_callk referenced in function "void __cdecl sol::detail::handle_protected_exception<1,class sol::basic_reference<0> >(struct lua_State *,class sol::optional<class std::exception const &>,char const *,struct sol::detail::protected_handler<1,class sol::basic_reference<0> > &)" (??$handle_protected_exception@$00V?$basic_reference@$0A@@sol@@@detail@sol@@YAXPEAUlua_State@@V?$optional@AEBVexception@std@@@1@PEBDAEAU?$protected_handler@$00V?$basic_reference@$0A@@sol@@@01@@Z)
main.cpp.obj : error LNK2019: unresolved external symbol __imp_lua_pcallk referenced in function "private: enum sol::call_status __cdecl sol::basic_protected_function<class sol::stack_reference,1,class sol::basic_reference<0> >::luacall<1>(__int64,__int64,struct sol::detail::protected_handler<1,class sol::basic_reference<0> > &)const " (??$luacall@$00@?$basic_protected_function@Vstack_reference@sol@@$00V?$basic_reference@$0A@@2@@sol@@AEBA?AW4call_status@1@_J0AEAU?$protected_handler@$00V?$basic_reference@$0A@@sol@@@detail@1@@Z)
main.cpp.obj : error LNK2019: unresolved external symbol __imp_luaopen_coroutine referenced in function "public: void __cdecl sol::state_view::open_libraries<enum sol::lib,enum sol::lib,enum sol::lib,enum sol::lib>(enum sol::lib &&,enum sol::lib &&,enum sol::lib &&,enum sol::lib &&)" (??$open_libraries@W4lib@sol@@W412@W412@W412@@state_view@sol@@QEAAX$$QEAW4lib@1@000@Z)
main.cpp.obj : error LNK2019: unresolved external symbol __imp_luaopen_utf8 referenced in function "public: void __cdecl sol::state_view::open_libraries<enum sol::lib,enum sol::lib,enum sol::lib,enum sol::lib>(enum sol::lib &&,enum sol::lib &&,enum sol::lib &&,enum sol::lib &&)" (??$open_libraries@W4lib@sol@@W412@W412@W412@@state_view@sol@@QEAAX$$QEAW4lib@1@000@Z)
main.cpp.obj : error LNK2019: unresolved external symbol __imp_luaL_requiref referenced in function "public: void __cdecl sol::state_view::open_libraries<enum sol::lib,enum sol::lib,enum sol::lib,enum sol::lib>(enum sol::lib &&,enum sol::lib &&,enum sol::lib &&,enum sol::lib &&)" (??$open_libraries@W4lib@sol@@W412@W412@W412@@state_view@sol@@QEAAX$$QEAW4lib@1@000@Z)
CppTesting.exe : fatal error LNK1120: 9 unresolved externals
ninja: build stopped: subcommand failed.
```

**Additional context**
I don't know if it's a specific header in STB causing this issue but when I manually added stb_image from github the linker errors were gone.
This does not seem to happen when luajit is used without sol2. Although using raw lua while sol2 is still linked still produces linker errors with a similar test cases.
e.g.
```
#include "lua.hpp"

int main(int argc, char* argv[])
{

	lua_State* L = luaL_newstate();

	// Load the standard libraries
	luaL_openlibs(L);

	// Run Lua code
	luaL_dostring(L, "print('Hello, World!')");

	// Close the Lua state
	lua_close(L);

	return 0;
}
```
I used CLion for this example. pkgconf is used to find the luajit files because there doesn't seem to be a find_package for luajit. sol2 is used because it makes using lua much easier. STB isn't used but its intended purpose for a larger project is the stb_image library.
Not sure if this format is what you want, if I've done something wrong, if I'm missing anything or if this isn't really related to vcpkg.

**Comment by MonicaLiu0311** [comment]
Confirm, repro locally.

```
PS G:\vcpkg> ./vcpkg version
vcpkg package management program version 2023-11-16-4c1df40a3c5c5e18de299a99e9accb03c2a82e1e

See LICENSE.txt for license information.
```

**Comment by MonicaLiu0311** [comment]
Fixed by #35545.
